File: bai_tap_lon/lib/models/lesson.dart
enum LessonType { vocabulary, grammar, communication, phrase, commonWords, pronunciation }

class Lesson {
  final String id;
  final String title;
  final String description;
  final LessonType type;
  final List<String> vocabularyIds; // For vocabulary lessons
  final List<String> phraseIds; // For phrase lessons
  final List<String> pronunciationIds; // For pronunciation lessons

  Lesson({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    this.vocabularyIds = const [],
    this.phraseIds = const [],
    this.pronunciationIds = const [],
  });
}

File: be/app/Models/Lesson.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Lesson extends Model
{
    protected $fillable = [
        'title',
        'description',
        'type',
        'vocabulary_ids',
        'phrase_ids',
        'exercise_ids',
        'pronunciation_ids',
    ];

    protected $casts = [
        'vocabulary_ids' => 'array',
        'phrase_ids' => 'array',
        'exercise_ids' => 'array',
        'pronunciation_ids' => 'array',
    ];

    public function vocabulary()
    {
        return $this->belongsToMany(Vocabulary::class, 'lesson_vocabulary');
    }

    public function phrases()
    {
        return $this->belongsToMany(Phrase::class, 'lesson_phrases');
    }

    public function exercises()
    {
        return $this->belongsToMany(Exercise::class, 'lesson_exercises');
    }

    public function quizzes()
    {
        return $this->hasMany(Quiz::class);
    }
}
